src/prefect/server/services/task_run_recorder.py

```diff
@@ -176,8 +177,39 @@ async def record_task_run_event(event: ReceivedEvent, depth: int = 0):
     )
 
 
+async def record_lost_follower_task_run_events():
+    events = await causal_ordering().get_lost_followers()
+
+    for event in events:
+        await record_task_run_event(event)
+
+
+async def periodically_process_followers(periodic_granularity: timedelta):
+    """Periodically process followers that are waiting on a leader event that never arrived"""
+    logger.info(
+        "Starting periodically process followers task every %s seconds",
+        periodic_granularity.total_seconds(),
+    )
+    while True:
+        try:
+            await record_lost_follower_task_run_events()
+        except asyncio.CancelledError:
+            logger.info("Periodically process followers task cancelled")
+            return
+        except Exception:
+            logger.exception("Error running periodically process followers task")
```

Comment:
Not sure if my suggestion is accurate, but I think this message could be clearer.
```suggestion
            logger.exception("Error while processing task-run-recorders followers.")
```